pages/glossary/programmatically-determined-context.md

```diff
@@ -0,0 +1,10 @@
+---
+title: Programmatically determined context
+key: programmatically-determined-context
+---
+
+The programmatically determined context of a piece of information (_e.g._ an element) is all the additional information that has a [programmatically determinable](https://www.w3.org/TR/WCAG21/#dfn-programmatically-determinable) [relationship](https://www.w3.org/TR/WCAG21/#dfn-relationships) to it.
+
+It includes, but is not restricted to, `aria-label` attribute, element that is referred by `aria-labelledby` or `aria-describedby` attributes, the `title` attribute, ancestor `label` elements, `label` elements referring the current element with a `for` attribute, ancestor elements, table header associated with the cell containing the information, headings, text inside the same paragraph (`p` element), sentence, table cell, list item, …
```

Comment:
I don't think this should be an open-ended definition (e.g. "includes but not restricted to"). There isn't an endless list of programatically determinable content. We should list all the programatically determined context that an element has, and be objective about it. So when you say "text inside the same paragraph", does that mean a child of, a descendant of the DOM tree, of the flat tree? What **exactly** does that mean?

We can be precise here, so I think we should be.